okio/src/commonMain/kotlin/okio/ByteString.kt

```diff
@@ -57,6 +57,20 @@ internal constructor(data: ByteArray) : Comparable<ByteString> {
   /** Returns this byte string encoded in hexadecimal.  */
   fun hex(): String
 
+  /** Returns the 128-bit MD5 hash of this byte string.  */
+  fun md5(): ByteString
+
+  /** Returns the 160-bit SHA-1 hash of this byte string.  */
+  fun sha1(): ByteString
+
+  /** Returns the 256-bit SHA-256 hash of this byte string.  */
+  fun sha256(): ByteString
+
+  /** Returns the 512-bit SHA-512 hash of this byte string.  */
+  fun sha512(): ByteString
+
+  internal fun digest(algorithm: String): ByteString
```

Comment:
This doesn't appear to be used from common code. The existing jvm/non-jvm versions can just be `private` in their respective classes.

Reply:
We can't do this. However, the function _does_ have to be open. I almost forgot about this being overridden in `SegmentedByteString`. I will be handling that in my next commit.